### PR TITLE
B2B Ecommerce Runbooks AB#38998

### PR DESCRIPTION
**Pull request by codapus-co**
Create runbooks for:
- Put Magento in Maintenance Mode and give Monsoon access for 2 hours on the var fileshare
- Remove Magento from Maintenance Mode
- Update the IP bypass files
- Disable Cron Jobs
- Re-enable Cron Jobs

### CODE REVIEW

File: external/m2/Disable-M2MaintenanceMode.ps1 [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+function Disable-M2MaintenanceMode {
+    
+    param (
+        [parameter(Mandatory = $true, Position = 0)]
+        [ValidateSet('dev', 'test', 'prod')]
+        [string] $Environment
+    )
+
+    $resourceGroupName = "b2b-ec-$Environment"
+    $storageAccountName = "unib2becop$Environment"
```
Comment (yavar-hasanov) [inline external/m2/Disable-M2MaintenanceMode.ps1:16]: can we expose these names as a parameter names/prefixes instead? The runbook operator can then pass the correct values when they start a runbook job.

File: external/m2/Enable-M2CronJobs.ps1 [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+function Enable-M2CronJobs {
+    
+    param (
+        [parameter(Mandatory = $true, Position = 0)]
+        [ValidateSet('dev', 'test', 'prod')]
+        [string] $Environment
+    )
+
+    $resourceGroupName = "b2b-ec-$Environment"
+    $storageAccountName = "unib2becop$Environment"
```
Comment (yavar-hasanov) [inline external/m2/Enable-M2CronJobs.ps1:16]: same here.

File: external/m2/Enable-M2MaintenanceMode.ps1 [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    }
+
+    # Reader
+    Grant-PIMAccess -UserGroupName "Monsoon Developers" -StorageAccountId $storageAccount.Id -RoleDefinitionId "acdd72a7-3385-48ef-bd42-f606fba81ae7" -DurationInHours 2
+
+    # Storage File Data SMB Share Reader
+    Grant-PIMAccess -UserGroupName "Monsoon Developers" -StorageAccountId $storageAccount.Id -RoleDefinitionId "aba4ae5f-2193-4029-9191-0cb91df5e314" -DurationInHours 2
+
+    # Storage File Data Privileged Reader
+    Grant-PIMAccess -UserGroupName "Monsoon Developers" -StorageAccountId $storageAccount.Id -RoleDefinitionId "b8eda974-7b85-4f76-af95-65846b26df6d" -DurationInHours 2
```
Comment (yavar-hasanov) [inline external/m2/Enable-M2MaintenanceMode.ps1:107]: possibly group names too.

File: external/m2/Set-M2MaintenanceModeAllowedIPs.ps1 [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        [parameter(Mandatory = $true, Position = 0)]
+        [ValidateSet('dev', 'test', 'prod')]
+        [string] $Environment,
+
+        [parameter(Mandatory = $true, Position = 1)]
+        [string] $AllowedIPs
+    )
+
+    $resourceGroupName = "b2b-ec-$Environment"
+    $storageAccountName = "unib2becop$Environment"
```
Comment (yavar-hasanov) [inline external/m2/Set-M2MaintenanceModeAllowedIPs.ps1:26]: same here.

**Review by yavar-hasanov — APPROVED** [review]

**Review by VaclavK — APPROVED** [review]

**Review by rmjoia — APPROVED** [review]

**Pull request merged**